IrohaCrypto/Classes/sr25519/SS58AddressFactory.m

```diff
@@ -49,12 +66,28 @@ - (nullable NSString*)addressFromPublicKey:(id<IRPublicKeyProtocol> _Nonnull)pub
     return [addressData toBase58];
 }
 
+- (UInt8) decodeTypeFromData:(NSData *)addressData {
+    UInt8 first = ((uint8_t*)addressData.bytes)[0];
```

Comment:
Looking to encoding we need to check rightmost 2 bits. If 01 extended and 00 regular.